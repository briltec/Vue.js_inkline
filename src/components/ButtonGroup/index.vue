<template src="./template.html" />
<script src="./script.js" />
<style src="../Button/style.styl" lang="stylus" scoped />
<style src="./style.styl" lang="stylus" scoped />
