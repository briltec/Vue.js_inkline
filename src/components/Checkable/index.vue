<script src="./script.js" />
<!--<style src="./style.styl" lang="stylus" />-->
