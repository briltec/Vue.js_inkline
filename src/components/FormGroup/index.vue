<template src="template.html" />
<script src="script.js" />
<style src="../Input/style.styl" lang="stylus" scoped />
<style src="style.styl" lang="stylus" scoped />
