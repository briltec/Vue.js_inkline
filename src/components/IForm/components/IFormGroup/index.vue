<template src="./template.html" />
<style src="./style.scss" lang="scss" />
<script src="./script.js" />
