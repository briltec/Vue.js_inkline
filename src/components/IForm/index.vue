<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="../IInput/style.scss" lang="scss" scoped />
<style src="./style.scss" lang="scss" scoped />
