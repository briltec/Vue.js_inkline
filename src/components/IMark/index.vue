<template src="./template.html" />
<script src="./script.ts" lang="ts" />
