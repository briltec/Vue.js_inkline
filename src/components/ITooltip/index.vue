<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="./style.scss" lang="scss" />
