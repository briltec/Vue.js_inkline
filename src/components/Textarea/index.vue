<template src="./template.html" />
<style src="../Input/style.styl" lang="stylus" />
<script src="./script.js" />
