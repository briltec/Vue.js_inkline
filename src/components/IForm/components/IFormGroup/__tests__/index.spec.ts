import { render } from '@testing-library/vue';
import { IFormGroup } from '@inkline/inkline/components';

describe('Components', () => {
    describe('IFormGroup', () => {
        const props = {
            name: 'form-group',
            color: 'light',
            size: 'md'
        };

        it('should be named correctly', () => {
            expect(IFormGroup.name).toEqual('IFormGroup');
        });

        it('should render correctly', () => {
            const wrapper = render(IFormGroup, { props });

            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should add classes based on props', () => {
                    const wrapper = render(IFormGroup, {
                        props: {
                            disabled: true,
                            readonly: true,
                            inline: true,
                            required: true,
                            ...props
                        }
                    });

                    expect(wrapper.container.firstChild).toHaveClass(
                        `-${props.color}`,
                        `-${props.size}`,
                        '-disabled',
                        '-readonly',
                        '-inline',
                        '-required'
                    );
                });
            });
        });
    });
});
