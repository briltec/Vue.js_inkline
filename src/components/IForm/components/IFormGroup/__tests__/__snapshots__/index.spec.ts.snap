// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components IFormGroup should render correctly 1`] = `
<div class="form-group -light -md" name="form-group" role="inputgroup">
  <div class="input-wrapper -light -md">
    <!--v-if-->
    <div class="input">
      <!--v-if--><input role="input" contenteditable="true" name="input" type="text" tabindex="1" aria-disabled="false" aria-readonly="false">
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>
`;
